Konyang Aerospace Forces Corvette

shaky-jake opened the pull request:
Adds a Konyang Aerospace Forces (KASF) corvette to the Haneunim sector with six player slots.

My first PR so (gently) scream at me if I screwed something up.

shaky-jake commented:
!review

Generalcamo commented on maps/away/ships/konyang/kasf_ship/kasf_ship.dm:
```diff
@@ -0,0 +1,133 @@
+/datum/map_template/ruin/away_site/kasf_corvette
+	name = "KASF Corvette"
+	description = "An older design of patrol corvette that saw it's fair share of service in it's golden days among the Xanu fleets, the Sai-class corvette would be considered obsolete by modern standards were it not retrofitted with newer weaponry, sensors, and other ship systems. In recent decades this class of ship has largely been mothballed by Xanu, but a large number were reactivated and donat ...
```
```suggestion
	description = "An older design of patrol corvette that saw its fair share of service in its golden days among the Xanu fleets, the Sai-class corvette would be considered obsolete by modern standards were it not retrofitted with newer weaponry, sensors, and other ship systems. In recent decades this class of ship has largely been mothballed by Xanu, but a large number were reactivated and donated when Konyang declared independence from Sol and joined the Coalition of Colonies. Now it serves an important role in the KASF, commonly seen combatting pirates and controlling the flow of refugees from the Wildlands."
```

Generalcamo commented on maps/away/ships/konyang/kasf_ship/kasf_ship.dm:
```diff
@@ -0,0 +1,133 @@
+
+/singleton/submap_archetype/kasf_corvette
+	map = "KASF Corvette"
+	descriptor = "An older design of patrol corvette that saw it's fair share of service in it's golden days among the Xanu fleets, the Sai-class corvette would be considered obsolete by modern standards were it not retrofitted with newer weaponry, sensors, and other ship systems. In recent decades this class of ship has largely been mothballed by Xanu, but a large number were reactivated and donate ...
```
```suggestion
	descriptor = "An older design of patrol corvette that saw its fair share of service in its golden days among the Xanu fleets, the Sai-class corvette would be considered obsolete by modern standards were it not retrofitted with newer weaponry, sensors, and other ship systems. In recent decades this class of ship has largely been mothballed by Xanu, but a large number were reactivated and donated when Konyang declared independence from Sol and joined the Coalition of Colonies. Now it serves an important role in the KASF, commonly seen combatting pirates and controlling the flow of refugees from the Wildlands."
```

Generalcamo commented on maps/away/ships/konyang/kasf_ship/kasf_ship.dm:
```diff
@@ -0,0 +1,133 @@
+/obj/effect/overmap/visitable/ship/kasf_corvette
+	name = "KASF Corvette"
+	class = "KASFV"
+	desc = "An older design of patrol corvette that saw it's fair share of service in it's golden days among the Xanu fleets, the Sai-class corvette would be considered obsolete by modern standards were it not retrofitted with newer weaponry, sensors, and other ship systems. In recent decades this class of ship has largely been mothballed by Xanu, but a large number were reactivated and donated when ...
```
```suggestion
	desc = "An older design of patrol corvette that saw its fair share of service in its golden days among the Xanu fleets, the Sai-class corvette would be considered obsolete by modern standards were it not retrofitted with newer weaponry, sensors, and other ship systems. In recent decades this class of ship has largely been mothballed by Xanu, but a large number were reactivated and donated when Konyang declared independence from Sol and joined the Coalition of Colonies. Now it serves an important role in the KASF, commonly seen combatting pirates and controlling the flow of refugees from the Wildlands."
```

Generalcamo commented on maps/away/ships/konyang/kasf_ship/kasf_ship.dm:
```diff
@@ -0,0 +1,133 @@
+	name = "KASF Shuttle"
+	class = "KASFV"
+	designation = "Kuoyu"
+	desc = "Somewhat bulky by modern standards, the aging Stalwart-class transport is a vessel being phased out of service in Xanu fleets to make way for newer, more efficient transports. Though considered obsolete compared to it's successor shuttles, this transport is still reliable enough that it's seen extensive use by Konyang's Aerospace Forces in recent years."
```
```suggestion
	desc = "Somewhat bulky by modern standards, the aging Stalwart-class transport is a vessel being phased out of service in Xanu fleets to make way for newer, more efficient transports. Though considered obsolete compared to its successor shuttles, this transport is still reliable enough that it's seen extensive use by Konyang's Aerospace Forces in recent years."
```

RustingWithYou commented on maps/away/ships/konyang/kasf_ship/kasf_ship.dm:
```diff
@@ -0,0 +1,133 @@
+	ship_cost = 1
+	id = "kasf_corvette"
+	shuttles_to_initialise = list(/datum/shuttle/autodock/overmap/kasf_shuttle)
+
```
```suggestion
	unit_test_groups = list(3)
```

DreamySkrell reviewed (changes requested):
![image](https://github.com/Aurorastation/Aurora.3/assets/107256943/aab17d3e-b65b-4bdc-bec2-9ef75ded2df7)

This is a big ship, just two tiny single-tile airlocks (marked in green) are not going to be enough.

There should be at least one more airlock near the front of the ship, and one at the rear (marked in red). The existing two airlocks should be made bigger, 2x2 tiles is fine.

There should be at least one proper docking airlock for the ship. Area marked in blue would be good I think, and should fit it nicely. Could even be another one on the opposite side so it's mirrored.

Please use the new airlock markers for any new airlocks. There are working examples on the Horizon, and there's the helper guidelines map too.

---

![image](https://github.com/Aurorastation/Aurora.3/assets/107256943/7b9b50d7-2dc3-4872-81ea-86476e58289a)

There should be one of these bigger pressure tank for propulsion, cause it's super annoying to have to change canisters often like this.

Also, no windows in front of engines - just put normal walls between engines and the pressurized interior. It's a really old mapping style that is just ugly and strange, and makes no sense. It's weird to see the engines working and burning fuel from... inside the ship. "Realistically", windows there would be a huge structural weak point too.

---

Other than the above things that need to be fixed, the ship looks cool, and I like it.

shaky-jake commented:
Since there's not any way to add fore or aft airlocks in a way that makes sense without tearing up existing areas, none will be added. The ship just wasn't really designed to accommodate for them, but the existing airlocks have been expanded now anyway.

All other requests have been addressed:
- Added two docking ports which can double as 2x1 airlocks
- Expanded existing 1x1 airlocks to 3x2
- Replaced glass behind thrusters with solid walls
- Added a 10,000L pressure tank to each thruster line

DreamySkrell reviewed (changes requested):
Same thing as on the IPC Refugee ship: 
> The exterior space landmarks are kinda too close to the ship - I'm pretty sure the Intrepid could only use the south-most one. Please either move them to be a bit further away from the ship, or what I think would be better, would be to just add two new landmarks that are named like "XYZ... (far)", and are further out, so shuttles that can use the closer landmarks have it more convenient, but other bigger shuttles can still have the other options.

![image](https://github.com/Aurorastation/Aurora.3/assets/107256943/f793886e-1fcb-435a-be12-9571442afd8c)
Also same as on the Refugee ship, the area on the left side is not needed.

As for the airlocks - ehh, it's not ideal, I would like more of them, but it's fine I guess. Please remember for future projects to include exterior access airlocks.

shaky-jake commented:
All required changes addressed:
- Shrunk transit area to accommodate just the shuttle transit landmark
- Added two far navpoints (port and starboard)

DreamySkrell reviewed (approved):
looks good now
